pkg/auth/handler/webapp/authflowv2/select_account.go

```diff
@@ -146,6 +137,15 @@ func (h *AuthflowV2SelectAccountHandler) ServeHTTP(w http.ResponseWriter, r *htt
 		suppressIDPSessionCookie = webSession.SuppressIDPSessionCookie
 		oauthProviderAlias = webSession.OAuthProviderAlias
 
+		// When x_suppress_idp_session_cookie is true, ignore IDP session cookie.
+		if suppressIDPSessionCookie {
+			session = nil
+		}
+		// Ignore any session that is not allow to be used here
+		if !oauth.ContainsAllScopes(oauth.SessionScopes(session), []string{oauth.PreAuthenticatedURLScope}) {
+			session = nil
+		}
+
```

Comment:
How does this move fix the bug?

Reply:
The bug was caused by moving the code into `BeforeHandle`, but I forgot to move these code. As a result `suppressIDPSessionCookie` is not assigned at the time `if suppressIDPSessionCookie ` is run.